experimental/dds/tree2/src/feature-libraries/sequence-field/compose.ts

```diff
@@ -907,30 +571,35 @@ export class ComposeQueue<T> {
 			newMark,
 		};
 	}
+
+	private dequeueLength(length: number): ComposeMarks<T> {
+		return {
+			baseMark: this.baseMarks.dequeueUpTo(length),
+			newMark: this.newMarks.dequeueUpTo(length),
+		};
+	}
 }
 
-type ComposeMarks<T> =
-	| {
-			baseMark: Mark<T>;
-			newMark: Mark<T>;
-			areInverses: true;
-	  }
-	| {
-			baseMark?: Mark<T>;
-			newMark?: Mark<T>;
-			areInverses?: false;
-	  };
+interface ComposeMarks<T> {
+	baseMark?: Mark<T>;
+	newMark?: Mark<T>;
+}
 
-/**
- * @returns true iff both reattaches target cells that were affected by the same detach.
- * The target cells may or may not overlap depending on detach index information.
- *
- * Only valid in the context of a compose (i.e., the output context of `baseMarks` is the input context of `newMark`).
- */
-function areRelatedReattaches<T>(baseMark: Reattach<T>, newMark: Reattach<T>): boolean {
-	const newEither = newMark.lastDetachedBy ?? newMark.detachedBy;
-	const baseEither = baseMark.lastDetachedBy ?? baseMark.detachedBy;
-	return newEither === baseEither;
+function areInverseMoves(
+	baseMark: MoveMark<unknown>,
+	baseIntention: RevisionTag | undefined,
+	newMark: MoveMark<unknown>,
+	newIntention: RevisionTag | undefined,
+): boolean {
+	if (
+		baseMark.type === "ReturnTo" &&
+		baseMark.detachEvent?.revision === (newMark.revision ?? newIntention)
```

Comment:
Shouln't this be
```suggestion
		(baseIntention ?? baseMark.detachEvent?.revision) === (newIntention ?? newMark.revision)
```

Reply:
No because `baseMark.detachEvent?.revision` is already an intention. Also I got confused with baseRevision.